src/test/java/bio/terra/app/controller/SnapshotsApiControllerTest.java

```diff
@@ -528,6 +548,107 @@ void deleteSnapshotBuilderSettingsForbidden() throws Exception {
     verifyAuthorizationCall(iamAction);
   }
 
+  @Test
+  void testGetConcepts() throws Exception {
+    SnapshotBuilderGetConceptsResponse expected = makeGetConceptsResponse();
+    when(snapshotBuilderService.getConceptChildren(SNAPSHOT_ID, CONCEPT_ID, TEST_USER))
+        .thenReturn(expected);
+    String actualJson =
+        mvc.perform(get(GET_CONCEPTS_ENDPOINT, SNAPSHOT_ID, CONCEPT_ID))
+            .andExpect(status().isOk())
+            .andReturn()
+            .getResponse()
+            .getContentAsString();
+    SnapshotBuilderGetConceptsResponse actual =
+        TestUtils.mapFromJson(actualJson, SnapshotBuilderGetConceptsResponse.class);
+    assertThat("Concept list and sql is returned", actual, equalTo(expected));
+
+    verifyAuthorizationCall(IamAction.READ_AGGREGATE_DATA);
+  }
+
+  @Test
+  void getSnapshotBuilderCount() throws Exception {
+    mockValidators();
+    var cohorts = List.of(SnapshotBuilderTestData.createCohort());
+    int count = 1234;
+    when(snapshotBuilderService.getCountResponse(SNAPSHOT_ID, cohorts, TEST_USER))
+        .thenReturn(
+            new SnapshotBuilderCountResponse()
+                .result(new SnapshotBuilderCountResponseResult().total(count)));
+    mvc.perform(
+            post(GET_COUNT_ENDPOINT, SNAPSHOT_ID)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(TestUtils.mapToJson(new SnapshotBuilderCountRequest().cohorts(cohorts))))
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$.result.total").value(count));
+    verifyAuthorizationCall(IamAction.READ_AGGREGATE_DATA);
+  }
+
+  private static Stream<String> searchTextArguments() {
```

Comment:
If you rename the source function to match the function it supplies, I think you can omit the argument to `@MethodSource`
```suggestion
  private static Stream<String> testSearchConcepts() {
```